Добавление юнит-тестов в файле tests.py для методов работы с сущностью  "книги"

alenaqa5 opened the pull request.

alenaqa5 committed:
Добавление юнит-тестов в файле tests.py для методов работы с сущностью "книги"

AlexKitIt commented on tests.py:
```diff
@@ -1,24 +1,68 @@
+    def test_get_books_genre_type_dict(self):
+        collector = BooksCollector()
+        collector.add_new_book('Ретро в конце спринта')
+        assert type(collector.get_books_genre()) == dict
```
Нужно исправить: этот метод проверяется по конкретному словарю

AlexKitIt commented on tests.py:
```diff
@@ -1,24 +1,68 @@
+        collector.add_book_in_favorites('На моей стороне не воспроизводится')
+        collector.add_new_book('Коллеги, микрофон забыл включить')
+        collector.add_book_in_favorites('Коллеги, микрофон забыл включить')
+        assert len(collector.favorites) == 2
```
Нужно исправить: не проверен метод get_list_of_favorites_books

alenaqa5 committed:
Исправление тестов:
 1. вместо проверки типа элемента словаря - проверяем получение жанра
 2. вместо получение длинны списка избранного - проверяем наличие добавленных книг

AlexKitIt commented on tests.py:
```diff
@@ -1,24 +1,69 @@
+        collector.add_book_in_favorites('На моей стороне не воспроизводится')
+        collector.add_new_book('Коллеги, микрофон забыл включить')
+        collector.add_book_in_favorites('Коллеги, микрофон забыл включить')
+        assert collector.favorites == ['На моей стороне не воспроизводится','Коллеги, микрофон забыл включить']
```
Нужно исправить: повторный коммент: где проверка метода get_list_of_favorites_books?? не атрибута а метода

alenaqa5 committed:
Исправление теста с получением списка избранного:
 1. вместо проверки атрибута favorites проверяю именно метод get_list_of_favorites_books
